scimma_admin/hopskotch_auth/views.py

```diff
@@ -482,6 +528,51 @@ def edit_topic(request):
         )
 
 
+@require_POST
+@login_required
+def change_topic_description(request):
+    logger.info(f"User {request.user.username} ({request.user.email}) to set the description for topic ID "
+                f"{request.POST.get('group_id','<unset>')} from {request.META['REMOTE_ADDR']}")
+
+    if not "topic_id" in request.POST or len(request.POST["topic_id"])==0:
+        return redirect_with_error(request, "Set a topic description",
+                                   "Missing or invalid topic ID", "index")
+    if "description" not in request.POST:
+        return redirect_with_error(request, "Set a topic description",
+                                   "Missing topic description", "index")
+
+    topic_id = request.POST["topic_id"]
+    try:
+        topic = KafkaTopic.objects.get(id=topic_id)
+    except ObjectDoesNotExist as dne:
+        return redirect_with_error(request, "Set a topic description",
+                                   "No such topic exists", "index")
+
+    # make sure the requesting user actually has the rights to modify this group
+    # the user must be a group owner or a staff member
+    if not is_group_owner(request.user,topic.owning_group.id) and not request.user.is_staff:
+        return redirect_with_error(request,
+                                   f"Set a description for topic {topic.name}",
+                                   "Requester is not an owning group owner or staff member",
```

Comment:
Is this the message presented to the user? It's pretty opaque to them, probably. Maybe something like "`This topic is owned by the {Group.get(id=topic.owning_group.id).name} group, and only the owner of that group can change the description`"

Just a thought, what you have is OK.